diff --git a/core/modules/language/src/ConfigurableLanguageManager.php b/core/modules/language/src/ConfigurableLanguageManager.php
index c4187aaf60..b72f9b3709 100644
--- a/core/modules/language/src/ConfigurableLanguageManager.php
+++ b/core/modules/language/src/ConfigurableLanguageManager.php
@@ -410,27 +410,25 @@ public function getLanguageSwitchLinks($type, Url $url) {
         if ($reflector->implementsInterface('\Drupal\language\LanguageSwitcherInterface')) {
           $original_languages = $this->negotiatedLanguages;
           $result = $this->negotiator->getNegotiationMethodInstance($method_id)->getLanguageSwitchLinks($this->requestStack->getCurrentRequest(), $type, $url);
+          $result = array_filter($result, function (array $link): bool {
+            $url = $link['url'] ?? NULL;
+            $language = $link['language'] ?? NULL;
+            if ($language instanceof LanguageInterface) {
+              $this->negotiatedLanguages[LanguageInterface::TYPE_CONTENT] = $language;
+              $this->negotiatedLanguages[LanguageInterface::TYPE_INTERFACE] = $language;
+            }
+            try {
+              return $url instanceof Url && $url->access();
+            }
+            catch (\Exception $e) {
+              return FALSE;
+            }
+          });
+          $this->negotiatedLanguages = $original_languages;
 
           if (!empty($result)) {
             // Allow modules to provide translations for specific links.
             $this->moduleHandler->alter('language_switch_links', $result, $type, $url);
-
-            $result = array_filter($result, function (array $link): bool {
-              $url = $link['url'] ?? NULL;
-              $language = $link['language'] ?? NULL;
-              if ($language instanceof LanguageInterface) {
-                $this->negotiatedLanguages[LanguageInterface::TYPE_CONTENT] = $language;
-                $this->negotiatedLanguages[LanguageInterface::TYPE_INTERFACE] = $language;
-              }
-              try {
-                return $url instanceof Url && $url->access();
-              }
-              catch (\Exception $e) {
-                return FALSE;
-              }
-            });
-            $this->negotiatedLanguages = $original_languages;
-
             $links = (object) ['links' => $result, 'method_id' => $method_id];
             break;
           }
